websites/H/Holodex/presence.ts

```diff
@@ -205,18 +197,37 @@ function getHomeFavsCategory() {
   }
 }
 
+/**
+ * Get the Category on the Channels Page
+ */
+function getChannelsCategory() {
+  switch ([].indexOf.call(document.querySelector("[role='tablist'] div.v-slide-group__content").children, document.querySelector("[role='tablist'] div.v-slide-group__content [aria-selected=true]"))) {
+    case 1:
+      return "VTuber";
+    case 2:
+      return "Subber";
+    case 3:
+      return "Favorites";
+    case 4:
+      return "Blocked";
+    default:
+      return "Unsupported Category"; // This should never occur, if it occurs it's a holodex.net bug
+  }
+}
+
 
 presence.on("UpdateData", async () => {
+  updateData.updateAll();
```

Comment:
I think you misunderstood me, I asked why you do all of these things in functions outside UpdateData

Reply:
It's a lot easier for a potential contributor to look at and understand
```js
updateData.updateAll();
```
and
```js
data.smallimage = await updateData.getSmallImage();
data.details = updateData.getDetails();
data.state = updateData.getState();
```
compared to a bunch of random switch cases and querySelector functions randomly placed in the UpdateData event (I just realized that updateData might be a poor name for my object as it might cause confusion, i'll rename it to dataUpdater or something so its dataUpdater.updateAll ).

Putting everything in their own seperate functions in a single object also allows the use of jsDoc, which vscode supports out of the box.

<details><summary>and you can collapse everything lol</summary>
<img src="https://user-images.githubusercontent.com/49831545/124818058-3a765a00-df73-11eb-904b-a8a542cb3e05.png" height="120px"></details>